tests/Controller/MediaControllerTest.php

```diff
@@ -138,29 +139,36 @@ public function testViewActionRendersView()
         $this->assertSame('renderResponse', $response->getContent());
     }
 
-    private function configureDownloadSecurity($pool, $media, $request, $isGranted)
-    {
+    private function configureDownloadSecurity(
+        ObjectProphecy $pool,
+        Media $media,
+        Request $request,
+        bool $isGranted
+    ): void {
         $strategy = $this->prophesize(DownloadStrategyInterface::class);
 
         $pool->getDownloadSecurity($media)->willReturn($strategy->reveal());
         $strategy->isGranted($media, $request)->willReturn($isGranted);
     }
 
-    private function configureGetMedia($id, $media)
+    private function configureGetMedia(int $id, ?Media $media): void
     {
         $mediaManager = $this->prophesize(BaseEntityManager::class);
 
         $this->container->get('sonata.media.manager.media')->willReturn($mediaManager->reveal());
         $mediaManager->find($id)->willReturn($media);
     }
 
-    private function configureGetProvider($pool, $media, $provider)
-    {
+    private function configureGetProvider(
+        ObjectProphecy $pool,
```

Comment:
You wouldnt add a rypehibt for Phpunit mock object here, why not type hinting the real class?

Reply:
Because it's not an instance of said class until you call `reveal()`

Reply:
I know maybe it confuses more then it helps but from a technical POV it’s fine

Reply:
Yeah I thought it was not great, but then thought other people would want to add a type hint and lose time finding out why it wasn't added. Plus reviewers would have asked why I did not put one :P Plus you get auto-complete in the method, so…